include/swift/Basic/ThreadSafeRefCounted.h

```diff
@@ -28,18 +28,19 @@ namespace swift {
 /// FIXME: This should eventually move to llvm.
 template <class Derived>
 class ThreadSafeRefCountedBase {
-  mutable llvm::sys::cas_flag ref_cnt;
+  mutable std::atomic<unsigned> ref_cnt;
 
 protected:
   ThreadSafeRefCountedBase() : ref_cnt(0) {}
 
 public:
   void Retain() const {
-    llvm::sys::AtomicIncrement(&ref_cnt);
+    ref_cnt.fetch_add(1, std::memory_order_relaxed);
   }
 
   void Release() const {
-    int refCount = static_cast<int>(llvm::sys::AtomicDecrement(&ref_cnt));
+    int refCount =
+        static_cast<int>(ref_cnt.fetch_sub(1, std::memory_order_relaxed));
```

Comment:
Are you sure?..  I'm confused -- the GCC docs say "In most cases, these builtins are considered a full barrier." for `__sync_add_and_fetch`.